app/views/rb_master_backlogs/index.html.erb

```diff
@@ -43,6 +43,16 @@ See doc/COPYRIGHT.rdoc for more details.
 <% html_title l(:label_backlogs) %>
 <%= toolbar title: l(:label_backlogs) %>
 
+<% if not params[:show_sub] or  params[:show_sub] != '1' %>
+  <%= link_to t(:backlogs_show_subprojects), url_for(params.merge(:show_sub => "1"))%>
+  </br>
+  </br>
+<% else %>
+  <%= link_to t(:backlogs_hide_subprojects), url_for(params.merge(:show_sub => "0"))%>
+  </br>
+  </br>
+<% end %>
```

Comment:
Using `url_for(params.merge(...))` risks having somebody inject parameters into the url with malicious intent. Actually, rails 5, to which we are migrating right now, will not let you do this at all.

The good news is, that the fix is more than easy. Your intention behind calling `url_for(params.merge(:show_sub => "0"))` can be implemented by simply leaving out the `params.merge` part so by writing `url_for(:show_sub => "0")` or `url_for(show_sub: "0")`. `url_for` will already take all the params and use them for generating the url. The difference is, it sanitizes the params.

Apart from that, this code would also benefit from having a helper method like `with_subprojects?` and by having the two code paths only differentiate by a parameter, e.g.:
```
<% show_sub_param = with_subprojects? ? "0" : "1" %>
<%= link_to t(:backlogs_show_subprojects), url_for(params.merge(show_sub: show_sub_param))%>
```

The `</br>` tags should be left out. It can be replaced by styling and positioning the link via css.